packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -199,11 +227,18 @@ class AttachCommand extends FlutterCommand {
       }
     } else {
       usesIpv6 = ipv6;
-      final int localPort = observatoryPort
-        ?? await device.portForwarder.forward(devicePort);
-      observatoryUri = usesIpv6
-        ? Uri.parse('http://[$ipv6Loopback]:$localPort/')
-        : Uri.parse('http://$ipv4Loopback:$localPort/');
+      if (debugUri != null) {
```

Comment:
This is taking on a life of its own, lets refactor it into a method

Reply:
Done.